### PR TITLE
Closes #987: TextEditor for htmlMarkupInputField.xhtml

### LINKED ISSUE #978: Explore possible new component for "Description" for WCAG compliance

### LINKED ISSUE #987: Implement p:textEditor for htmlMarkupInputField
Follow up on #978

### PR DESCRIPTION
**Pull request by dbojanek**

### CODE REVIEW

File: dataverse-webapp/src/main/webapp/resources/js/dv_rebind_bootstrap_ui.js [hunk 1]
```diff
@@ -336,6 +336,133 @@ function reinitializePrimefacesComponentsJS() {
            }
         });
     }
+
+    if(PrimeFaces.widget.TextEditor) {
```
Comment (wtr-icm) [inline dataverse-webapp/src/main/webapp/resources/js/dv_rebind_bootstrap_ui.js:340]: Why we need this additional config? Shouldn't the editor work out-of-the-box?

**Review by wtr-icm — COMMENTED** [review]
There is a note in PrimeFaces docs:

> TextEditor requires the owasp-java-html-sanitizer dependency added to your pom.xml to prevent XSS.

Shouldn't we sanitize in some way editor's input as it's described? XSS is one of the most common ways to attack websites.

File: dataverse-webapp/src/main/webapp/resources/js/dv_rebind_bootstrap_ui.js (hunk 1 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/webapp/resources/js/dv_rebind_bootstrap_ui.js:340]: Precisely that - the out-of-the-box behavior is not wcag compatible so we needed to change it. That was the main goal of this task.

**Comment by dbojanek** [comment]
I had a bit of a problem with Intellij not resolving sanitazer's dependency so for explore I've just made textEditor `secure=false` and have forgotten all about it.

**Review by wtr-icm — APPROVED** [review]

**Pull request merged**